File: Backend/Microservices/Admin/Eefa.Admin.Domain/Entities/Employee.cs
using System;

[HasUniqueIndex]
public partial class Employee : AuditableEntity
{
    public int PersonId { get; set; } = default!;
    public int UnitPositionId { get; set; } = default!;
    [UniqueIndex]
    public string EmployeeCode { get; set; } = default!;
    public DateTime EmploymentDate { get; set; } = default!;
    public bool Floating { get; set; } = default!;
    public DateTime? LeaveDate { get; set; }


    public virtual User CreatedBy { get; set; } = default!;
    public virtual User? ModifiedBy { get; set; } = default!;
    public virtual Role OwnerRole { get; set; } = default!;
    public virtual Person Person { get; set; } = default!;
    public virtual UnitPosition UnitPosition { get; set; } = default!;
}

File: Frontend/angular-app/src/app/modules/admin/entities/employee.ts
import {Person} from "./person";

export class Employee {
  public id!:number;
  public personId!:number;
  public nationalNumber!: string;
  public firstName!:string;
  public lastName!:string;
  public accountReferenceCode!: string;
  public branchId!:number;
  public unitId!:number;
  public unitPositionId!:number;
  public employeeCode!:number;
  public employmentDate!:string;
  public leaveDate!:string;
  public person!:Person;
  public legalTitle!:string;
  public governmentalTitle!:string;
}
